meshmode/array_context.py

```diff
@@ -437,77 +437,73 @@ def _get_array_container_ac_arith(ary: ArrayContainerWithArithmetic):
 
 # {{{ dataclass containers
 
-# FIXME: Document what this is good for
+# TODO Will this interfere with type-checking, should we want it later?
+# We could always change the definition based on typing.TYPE_CHECKING...
```

Comment:
What do you think about? It's a clunky workaround for not being able to tell from the type whether something is an array container.

Reply:
I'm not liking how hardcoded this `_ARRAY_CONTAINER_LIKE` thing is as opposed to everything else that's nicely extensible.

* Maybe we can change how `is_array_container` works? Just have a `register_as_array_container(cls)` that appends to `_ARRAY_CONTAINER_LIKE` and then `is_array_container` can just be `return issubclass(ary, _ARRAY_CONTAINER_LIKE)`?
* Can we use `f.type in is_array_container.registry`?

Reply:
b8c3c52